src/firefly/js/visualize/region/RegionFactory.js

```diff
@@ -74,37 +77,80 @@ export class RegionFactory {
      * @returns {array} an array of Region object
      */
     static parseRegionDS9(regionData, bAllowHeader = true, stopAt) {
-        var regionLines = regionData.reduce ((prev, oneLine) => {
-                var resRegions = oneLine.split(';');
+        const sep = ';';
+        const dLeft = ['{', '"', '\''];
+        const dRight = ['}', '"', '\''];
+
+        // collect region lines and each line may contain coordinate, region description or both
+        var regionLines = regionData.reduce( (rLines, oneLine) => {
+
```

Comment:
This is a lot of code for a reduce function.  I think you should break it up so that it is a little easier to follow. I see that the old approach was a reduce as well but it was not as big.